File: data/gpt-4o/readable_format_HPEC/fortran/file203.f90
subroutine init_array(n, a)
implicit none

DATA_TYPE, dimension(n, n) :: a
integer :: n
integer :: i,j
do i = 1, n
do j = 1, n
a(j, i) = ((DBLE(i - 1) * DBLE(j + 1)) + 2.0D0) / n
end do
end do
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file203.cpp
void init_array(int n, double a[n][n]) 
{
 for(int i = 0; i < n; i++)
 for(int j = 0; j < n; j++)
 a[j][i] = ((double(i - 1) * double(j + 1)) + 2.0) / n;
}
